### PR TITLE
[IMP] pos*: format files and add eslint test

### PR DESCRIPTION
**Pull request by vlst-odoo**
In this commit we format all the js files in the pos codebase and we add
a test that checks that the files are correctly formatted.

We use the eslint config file found at `web/tooling/_eslintrc.json`.

This is done because file formatting turns out to present a lot of
issues while writing code. One issue that makes us waste a lot of time
because of formatting is when someone accidentally formats a file while
working on it. The review becomes much more difficult, because the
reviewer has to figure out which lines were changed intentionally and
which were not.
This commit will prevent this issue from happening in the future,
because all files that are merged in the codebase will be formatted
already.

Task: 3750944
https://github.com/odoo/enterprise/pull/56088

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by vlst-odoo** [comment]
Hello @odoo/rd-security , can you please take a look at this?

File: odoo/addons/test_lint/tests/test_eslint.py [hunk 1]
```diff
@@ -39,3 +40,22 @@ def test_eslint(self):
         cmd = [eslint, '--no-eslintrc', '-c', eslintrc_path] + files_to_check
         process = subprocess.run(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE, check=False)
         self.assertEqual(process.returncode, 0, msg=process.stdout.decode())
+
+    def test_eslint_pos(self):
```
[the remark this answers is not in the document]
Comment (vlst-odoo) [reply odoo/addons/test_lint/tests/test_eslint.py:44]: The files in pos are formatted with the eslint config from `web/tooling`, which has stricter rules than the eslint file against which we check all js files.

File: odoo/addons/test_lint/tests/lint_case.py [hunk 2]
```diff
@@ -11,11 +11,11 @@ class LintCase(BaseCase):
     """ Utility method for lint-type cases
     """
 
-    def iter_module_files(self, *globs):
+    def iter_module_files(self, *globs, modules=get_modules()):
```
[the remark this answers is not in the document]
Comment (vlst-odoo) [reply odoo/addons/test_lint/tests/lint_case.py:14]: what do you mean?

File: odoo/addons/test_lint/tests/test_eslint.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (vlst-odoo) [reply odoo/addons/test_lint/tests/test_eslint.py:44]: @xmo-odoo Do you think that there might be a better way to test against the stricter esilnt config?

File: odoo/addons/test_lint/tests/lint_case.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (vlst-odoo) [reply odoo/addons/test_lint/tests/lint_case.py]: I understand. Do you think the change from the new commit will solve this issue?

File: odoo/addons/test_lint/tests/test_eslint.py [hunk 3]
```diff
@@ -39,3 +40,22 @@ def test_eslint(self):
...
+        modules = [module for module in get_modules() if re.match('point_of_sale|pos_.*|.*_pos|.*_pos_.*', module)]
+        files_to_check = [
+            p for p in self.iter_module_files('**/static/**/*.js', modules=modules)
+            if not re.match('.*/libs?/.*', p)  # don't check libraries
+        ]
+        eslintrc_path = file_path('web/tooling/_eslintrc.json')
+
+        _logger.info('Testing %s js files', len(files_to_check))
+        cmd = [eslint, '--no-ignore', '--no-eslintrc', '-c', eslintrc_path] + files_to_check
+        process = subprocess.run(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE, check=False)
```
[the remark this answers is not in the document]
Comment (vlst-odoo) [reply odoo/addons/test_lint/tests/test_eslint.py]: slight issue: 
image
@xmo-odoo Should i add back the `check` ?

[the remark this answers is not in the document]
Comment (vlst-odoo) [reply odoo/addons/test_lint/tests/test_eslint.py]: i added back the `check` in the meantime, you can see it in this old build https://runbot.odoo.com/runbot/batch/1351864/build/58582611

**Review by adgu-odoo — APPROVED** [review]
@robodoo r+

**Pull request closed without merging**